client/landing/stepper/declarative-flow/internals/steps-repository/domains/index.tsx

```diff
@@ -267,7 +267,8 @@ const DomainsStep: Step = function DomainsStep( { navigation, flow } ) {
 	const onUseYourDomainClick = ( domain?: string ) => {
 		if ( domain && isHundredYearDomainFlow( flow ) ) {
 			const leaveFlowFunction = exitFlow ?? window.location.assign;
-			leaveFlowFunction( `/setup/hundred-year-domain-transfer?new=${ domain }` );
+			leaveFlowFunction( `/setup/hundred-year-domain-transfer/domains?new=${ domain }` );
+			return;
```

Comment:
Fixes the first bug described in this PR - `Use your domain` component being briefly visible.